flink-connector-aws-e2e-tests/flink-connector-aws-kinesis-streams-e2e-tests/src/test/java/org/apache/flink/connector/kinesis/testutils/AWSKinesisResourceManager.java

```diff
@@ -0,0 +1,106 @@
+package org.apache.flink.connector.kinesis.testutils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DeleteStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamResponse;
+import software.amazon.awssdk.services.kinesis.model.StreamMode;
+import software.amazon.awssdk.services.kinesis.model.StreamModeDetails;
+import software.amazon.awssdk.services.kinesis.model.StreamSummary;
+import software.amazon.awssdk.services.kinesis.paginators.ListStreamsIterable;
+import software.amazon.awssdk.services.kinesis.waiters.KinesisWaiter;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/** Resource manager to create and clean up Kinesis Data Stream resources used for each test. */
+public class AWSKinesisResourceManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSKinesisResourceManager.class);
+
+    private static final String KINESIS_DATA_STREAMS_NAME_PREFIX = "flink-test-";
+    private static final int DEFAULT_SHARD_COUNT = 1;
+    private final KinesisClient kinesisClient;
+    private final KinesisWaiter kinesisWaiter;
+    private final Set<String> kinesisDataStreamARNs = new HashSet<>();
+
+    public AWSKinesisResourceManager(KinesisClient kinesisClient) {
+        this.kinesisClient = kinesisClient;
+        this.kinesisWaiter = KinesisWaiter.builder().client(kinesisClient).build();
+    }
+
+    public String createKinesisDataStream() {
+        String streamName = KINESIS_DATA_STREAMS_NAME_PREFIX + UUID.randomUUID();
+        LOGGER.info("Creating Kinesis Data Stream with name {}.", streamName);
+        kinesisClient.createStream(
+                CreateStreamRequest.builder()
+                        .streamName(streamName)
+                        .shardCount(DEFAULT_SHARD_COUNT)
+                        .streamModeDetails(
+                                StreamModeDetails.builder()
+                                        .streamMode(StreamMode.PROVISIONED)
+                                        .build())
+                        .build());
+        DescribeStreamResponse response =
+                kinesisClient.describeStream(
+                        DescribeStreamRequest.builder().streamName(streamName).build());
+        String streamARN = response.streamDescription().streamARN();
+        kinesisDataStreamARNs.add(streamARN);
+        LOGGER.info("Successfully created Kinesis Data Stream with ARN {}.", streamARN);
+
+        LOGGER.info("Waiting until Kinesis Data Stream with ARN {} is ACTIVE.", streamARN);
+        kinesisWaiter.waitUntilStreamExists(
```

Comment:
Just checking this waits for stream to be ACTIVE? It is not too clear from the [docs](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/kinesis/waiters/KinesisWaiter.html#waitUntilStreamExists(java.util.function.Consumer,java.util.function.Consumer))

Reply:
Yep we can see this from the [API itself](https://github.com/aws/aws-sdk-java-v2/blob/fa3fccae0818578633f09f6a9328ade04b7503d0/services/kinesis/src/main/resources/codegen-resources/waiters-2.json#L8-L14)